pydantic/env_settings.py

```diff
@@ -44,19 +45,58 @@ def _build_values(
         _env_file_encoding: Optional[str] = None,
         _secrets_dir: Union[Path, str, None] = None,
     ) -> Dict[str, Any]:
+        extra_settings = [source(self) for source in reversed(self.__config__.extra_settings_sources)]
         return deep_update(
-            self._build_secrets_files(_secrets_dir), self._build_environ(_env_file, _env_file_encoding), init_kwargs
+            *extra_settings,
+            self._build_secrets_files(_secrets_dir),
+            self._build_environ(_env_file, _env_file_encoding),
+            init_kwargs,
         )
 
+    def filter_relevant_env_vars(self, env_vars: Mapping[str, Optional[str]]) -> Dict[str, Optional[str]]:
```

Comment:
same with most of the methods below.

Reply:
can you move this down to where the code was before? If so hopefully we can reduce the number of changes and keep history easy to track.

Reply:
I'm going to revert this modification as the new implementation will only use `load_env_vars_from_source`.
Also, `load_env_vars_from_source` will be made private to avoid conflicts as you requested.